File: stacks_and_queues/stack_example.cpp
#include <gtest/gtest.h>

#include <stack>

TEST(StackExampleTest, TestStack)
{
    std::stack<int> stack {};

    stack.push(1);
    stack.push(2);
    stack.push(3);

    EXPECT_EQ(3, stack.top());
    stack.pop();

    EXPECT_EQ(2, stack.top());
    stack.pop();

    EXPECT_EQ(1, stack.top());
    stack.pop();

    stack.push(5);
    EXPECT_FALSE(stack.empty());
}

File: stacks_and_queues/stack_example.py
stack = []

stack.append(1)
stack.append(2)
stack.append(3)

print(stack.pop())
print(stack.pop())
print(stack.pop())

stack.append(5)

if not stack:
    print("Stack is empty!")
else:
    print(f"Stack is not empty, top is: {stack[-1]}")